j1939/Dm14Server.py

```diff
@@ -0,0 +1,283 @@
+            return
+
+        length = min(data[0], len(data) - 1)
+        # assert object_count == self.object_count
+        self.mem_data = data[1 : length + 1]
+        self.data_queue.put(data)
+        self._ca.unsubscribe(self._parse_dm16)
+        self._ca.subscribe(self.parse_dm14)
+        self.state = ResponseState.SEND_OPERATION_COMPLETE
+        self._send_dm15()
+
+    def bytes_to_int(self, data: bytearray) -> int:
+        """
+        Convert bytesaray to integer
+        :param bytearray data: bytearray to be converted to integer
+        """
+        return int.from_bytes(data, byteorder="little", signed=False)
+
+    def set_busy(self, busy: bool) -> None:
+        """
+        Sets busy variable to indicate if busy or not
+        :param busy: busy value
+        """
+        self._busy = busy
+
+    def generate_seed(self) -> int:
+        """
+        Generte a random seed value for key generation
+        """
+        seed = secrets.randbits(16)
+        if (seed == 0xFFFF) or (seed == 0x0000):
+            seed = 0xBEEF
+        return seed
+
+    def set_seed_key_algorithm(self, algorithm: callable) -> None:
+        """
+        Set seed key algorithm to be used for key generation
+        :param callable algorithm: seed-key algorithm
+        """
+        self._key_from_seed = algorithm
+
+    def set_seed_generator(self, algorithm: callable) -> None:
+        """
+        Sets seed generation algorithm to be used for generating a seed value
+        :param callable algorithm: seed generation algorithm
+        """
+        self._seed_generator = algorithm
+
+    def respond(
+        self,
+        proceed: bool,
+        data=None,
+        error: int = 0xFFFFFF,
+        edcp: int = 0xFF,
+    ) -> list:
+        """
+        Respond to DM14 query with the requested data or confimation of operation is good to proceed
+        :param bool proceed: whether the operation is good to proceed
+        :param list data: data to be sent to device
+        :param int error: error code to be sent to device
+        :param int edcp: value for edcp extension
+        """
+        if data is None:
+            data = []
+        self.proceed = proceed
+        self.data = data
+        self.error = error
+        self.edcp = edcp
+        self.status = (
+            j1939.Dm15Status.PROCEED.value
+            if proceed
+            else j1939.Dm15Status.OPERATION_FAILED.value
+        )
+        if self.status == j1939.Dm15Status.PROCEED.value:
+            self.state = ResponseState.SEND_PROCEED
+        else:
+            self.state = ResponseState.SEND_ERROR
+        # self._ca.unsubscribe(self._parse_dm14)
+        self._wait_for_data()
+        mem_data = None
+        if self.state == ResponseState.WAIT_FOR_DM16:
+            mem_data = self.data_queue.get(block=True, timeout=3)
+
+        return self.mem_data if self.mem_data is not None else None
```

Comment:
Does the if/else statement here do anything?

Reply:
No, not really I'll eliminate it